src/core/ui/context/login-methods.tsx

```diff
@@ -63,18 +81,84 @@ const LoginMethodsProvider = (
 };
 
 const useLoginMethods = () => {
-  const {
+  const { loginMethods, selectedLoginMethod, setSelectedLoginMethod } =
+    _useLoginMethods();
+  const { excludeLoginMethodTypes, includeLoginMethodTypes, viewOnly } =
+    useSignInContext();
+  const user = useUser();
+
+  const web3 = useMemo(() => {
+    console.log(
+      loginMethods,
+      excludeLoginMethodTypes,
+      includeLoginMethodTypes,
+      user?.loginMethods,
+      viewOnly
+    );
+    if (!viewOnly && user?.loginMethods.includes(LoginMethodType.Web3)) {
+      console.log('user');
+      // User is already logged in using web3 so we should not expose this.
+      return [];
+    }
+
+    if (excludeLoginMethodTypes?.includes(LoginMethodType.Web3)) {
+      console.log('exclude');
+      return [];
+    }
+
+    if (
+      includeLoginMethodTypes &&
+      !includeLoginMethodTypes?.includes(LoginMethodType.Web3)
+    ) {
+      console.log('include');
+      return [];
+    }
+
+    console.log('filter');
+    return loginMethods.filter(
+      (m) => m.type === LoginMethodType.Web3
+    ) as unknown as Web3LoginMethod[];
+  }, [
     loginMethods,
-    setLoginMethods,
-    selectedLoginMethod,
-    setSelectedLoginMethod,
-  } = _useLoginMethods();
+    excludeLoginMethodTypes,
+    includeLoginMethodTypes,
+    user?.loginMethods,
+    viewOnly,
+  ]);
 
-  return {
+  const web2 = useMemo(() => {
+    const loggedInMethods = user?.loginMethods || [];
+
+    return loginMethods
+      .filter((m) => m.id !== 'magic-link')
```

Comment:
Modified this to remove magic-link since it will be implemented in: https://linear.app/slashauth/issue/SLA-1843/sr-new-design-for-sign-in-with-email-magic-link